authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -91,9 +94,42 @@ protected RangerAuthorizationPlugin(String metalake, Map<String, String> config)
     rangerAdminName = config.get(RangerAuthorizationProperties.RANGER_USERNAME);
     // Apache Ranger Password should be minimum 8 characters with min one alphabet and one numeric.
     String password = config.get(RangerAuthorizationProperties.RANGER_PASSWORD);
-    rangerServiceName = config.get(RangerAuthorizationProperties.RANGER_SERVICE_NAME);
+
+    String serviceName = config.get(RangerAuthorizationProperties.RANGER_SERVICE_NAME);
     rangerClient = new RangerClientExtension(rangerUrl, authType, rangerAdminName, password);
 
+    if (Boolean.parseBoolean(
+        config.get(RangerAuthorizationProperties.RANGER_SERVICE_CREATE_IF_ABSENT))) {
+      if (serviceName == null) {
+        serviceName = config.get(BaseAuthorization.UUID);
+      }
+
+      try {
+        rangerClient.getService(serviceName);
+      } catch (RangerServiceException rse) {
+        if (rse.getStatus().equals(ClientResponse.Status.NOT_FOUND)) {
```

Comment:
Are we sure the status is 404 rather than a 400?
I still remember that `getRole()` returns 400 if a role is not found.

Reply:
Yes, they are different. I used the UT to verify this.